crates/ra_lsp_server/src/cargo_check.rs

```diff
@@ -0,0 +1,1316 @@
+//! cargo_check provides the functionality needed to run `cargo check` or
+//! another compatible command (f.x. clippy) in a background thread and provide
+//! LSP diagnostics based on the output of the command.
```

Comment:
This module/crate would benefit from discussing our threading/scheduling/cancellation model. Like, how many threads do we use them, when we fire them up, how we make sure they are cleanly joined.